### PR TITLE
[navigator] disable 'File: Delete' for root

### LINKED ISSUE #3593: [navigator] remove 'delete' quick-open commands and better handle keybinding
Currently, there is missing handling when performing `delete` by either the quick-open command or its keybinding since we do not prevent deleting the root which is troublesome and prone to many bugs and errors (example: delete root, and create new file).

The `context-menu` works correctly and prevents deleting the root, but the keybinding does not so we need to align.

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Fixes #3593

- Added missing `File:Delete` handling for `isEnabled`.
- We can no longer `delete` the root directory which is now consistent with the `FileNavigator` context-menu.
- Fixes mutliple issues for when a user proceeds to `delete` the root directory and perform various subsequent actions.

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/workspace/src/browser/workspace-delete-handler.ts [hunk 1]
```diff
@@ -37,6 +37,10 @@ export class WorkspaceDeleteHandler implements UriCommandHandler<URI[]> {
         return !!uris.length && !this.isRootDeleted(uris);
     }
 
+    isEnabled(uris: URI[]): boolean {
+        return !!uris.length && !this.isRootDeleted(uris);
+    }
+
     protected isRootDeleted(uris: URI[]): boolean {
```
Comment (elaihau) [inline packages/workspace/src/browser/workspace-delete-handler.ts]: This line is not part of this change but the function should be renamed imo. 

What it really means is "can this folder be deleted". i suggested to renamed it into "canDelete" or "isRoot". what do you think?

Comment (vince-fugnitto) [reply packages/workspace/src/browser/workspace-delete-handler.ts]: I agree, `isRootDeleted` should probably be rename to something more meaningful. When making the change I was confused on what it actually achieved.
